# Day 70 - Merging

1. Using the given products DataFrame:

  product_id          name    price
0        001  mobile phone  1490.00
1        004        laptop  3400.00
2        007         mouse    59.99
3        010        tablet   999.00

In [1]:
import pandas as pd
 
 
products_dict = {
    'product_id': ['001', '004', '007', '010'],
    'name': ['mobile phone', 'laptop', 'mouse', 'tablet'],
    'price': [1490.0, 3400.0, 59.99, 999.00]
}
 
orders_day_1_dict = {
    'order_id': ['1001', '1002', '1003', '1004'],
    'product_id': ['004', '001', '001', '007'],
    'quantity': [2, 1, 1, 3]
}       
 
orders_day_2_dict = {
    'order_id': ['1005', '1006', '1007'],
    'product_id': ['010', '001', '007'],
    'quantity': [2, 1, 1]
}                
 
products = pd.DataFrame(products_dict)
orders_day_1 = pd.DataFrame(orders_day_1_dict)
orders_day_2 = pd.DataFrame(orders_day_2_dict)
orders = orders_day_1.append(orders_day_2, ignore_index=True)
 
result = pd.merge(orders, products, on='product_id')
print(result)

  order_id product_id  quantity          name    price
0     1001        004         2        laptop  3400.00
1     1002        001         1  mobile phone  1490.00
2     1003        001         1  mobile phone  1490.00
3     1006        001         1  mobile phone  1490.00
4     1004        007         3         mouse    59.99
5     1007        007         1         mouse    59.99
6     1005        010         2        tablet   999.00


C:\Users\JAI GANESH\AppData\Local\Temp\ipykernel_18628\3083480874.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  orders = orders_day_1.append(orders_day_2, ignore_index=True)


2. The result DataFrame is given below:

  order_id product_id  quantity          name    price
0     1001        004         2        laptop  3400.00
1     1002        001         1  mobile phone  1490.00
2     1003        001         1  mobile phone  1490.00
3     1006        001         1  mobile phone  1490.00
4     1004        007         3         mouse    59.99
5     1007        007         1         mouse    59.99
6     1005        010         2        tablet   999.00

Based on the quantity and price columns, calculate a new column 'total' that stores the total amount for each order as shown below.
In response, print a result DataFrame to the console.

In [2]:
import pandas as pd


products_dict = {
    'product_id': ['001', '004', '007', '010'],
    'name': ['mobile phone', 'laptop', 'mouse', 'tablet'],
    'price': [1490.0, 3400.0, 59.99, 999.00]
}

orders_day_1_dict = {
    'order_id': ['1001', '1002', '1003', '1004'],
    'product_id': ['004', '001', '001', '007'],
    'quantity': [2, 1, 1, 3]
}       

orders_day_2_dict = {
    'order_id': ['1005', '1006', '1007'],
    'product_id': ['010', '001', '007'],
    'quantity': [2, 1, 1]
}                

products = pd.DataFrame(products_dict)
orders_day_1 = pd.DataFrame(orders_day_1_dict)
orders_day_2 = pd.DataFrame(orders_day_2_dict)
orders = orders_day_1.append(orders_day_2, ignore_index=True)
result = pd.merge(orders, products, on='product_id')

result['total'] = result['quantity'] * result['price']
print(result)

  order_id product_id  quantity          name    price    total
0     1001        004         2        laptop  3400.00  6800.00
1     1002        001         1  mobile phone  1490.00  1490.00
2     1003        001         1  mobile phone  1490.00  1490.00
3     1006        001         1  mobile phone  1490.00  1490.00
4     1004        007         3         mouse    59.99   179.97
5     1007        007         1         mouse    59.99    59.99
6     1005        010         2        tablet   999.00  1998.00


C:\Users\JAI GANESH\AppData\Local\Temp\ipykernel_18628\4015244350.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  orders = orders_day_1.append(orders_day_2, ignore_index=True)


3. The following result DataFrame is given:

  order_id product_id  quantity          name    price    total
0     1001        004         2        laptop  3400.00  6800.00
1     1002        001         1  mobile phone  1490.00  1490.00
2     1003        001         1  mobile phone  1490.00  1490.00
3     1006        001         1  mobile phone  1490.00  1490.00
4     1004        007         3         mouse    59.99   179.97
5     1007        007         1         mouse    59.99    59.99
6     1005        010         2        tablet   999.00  1998.00

Save this DataFrame to the following files:

'result.csv' -> skip index
'result.html'
'result.json' -> set indent to 4 spaces

In [3]:
import pandas as pd
 
 
products_dict = {
    'product_id': ['001', '004', '007', '010'],
    'name': ['mobile phone', 'laptop', 'mouse', 'tablet'],
    'price': [1490.0, 3400.0, 59.99, 999.00]
}
 
orders_day_1_dict = {
    'order_id': ['1001', '1002', '1003', '1004'],
    'product_id': ['004', '001', '001', '007'],
    'quantity': [2, 1, 1, 3]
}       
 
orders_day_2_dict = {
    'order_id': ['1005', '1006', '1007'],
    'product_id': ['010', '001', '007'],
    'quantity': [2, 1, 1]
}                
 
products = pd.DataFrame(products_dict)
orders_day_1 = pd.DataFrame(orders_day_1_dict)
orders_day_2 = pd.DataFrame(orders_day_2_dict)
orders = orders_day_1.append(orders_day_2, ignore_index=True)
result = pd.merge(orders, products, on='product_id')
result['total'] = result['quantity'] * result['price']
 
result.to_csv('result.csv', index=False)
result.to_html('result.html')
result.to_json('result.json', indent=4)

C:\Users\JAI GANESH\AppData\Local\Temp\ipykernel_18628\2758402380.py:25: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  orders = orders_day_1.append(orders_day_2, ignore_index=True)
